dmoj/judgeenv.py

```diff
@@ -347,13 +351,22 @@ def get_supported_problems_and_mtimes():
         A list of all problems in tuple format: (problem id, mtime)
     """
     problems = []
-    for dir in get_problem_roots():
-        if not os.path.isdir(dir):  # we do this check in case a problem root was deleted but persists in cache
-            continue
-        for problem in os.listdir(dir):
-            problem = utf8text(problem)
-            if os.access(os.path.join(dir, problem, 'init.yml'), os.R_OK):
-                problems.append((problem, os.path.getmtime(os.path.join(dir, problem))))
+    if problem_globs:
+        for dir_glob in problem_globs:
+            for problem_config in glob.iglob(os.path.join(dir_glob, 'init.yml')):
+                if os.access(problem_config, os.R_OK):
+                    problem_dir = os.path.dirname(problem_config)
+                    problem = utf8text(os.path.basename(problem_dir))
+                    problems.append((problem, os.path.getmtime(problem_dir)))
+
+    else:
+        for dir in get_problem_roots():
```

Comment:
Does this go away if you rebase? I thought we got rid of this code.

Reply:
You're thinking of the currently unmerged #1102.